optics-core/src/Optics/Setter.hs

```diff
@@ -102,14 +102,28 @@ over o = \f -> runFunArrow $ getOptic (castOptic @A_Setter o) (FunArrow f)
 -- 'over' is used, because the first coordinate of a pair is never forced.
 --
 over'
```

Comment:
The haddocks above here have another "TODO DOC" it would be great to nail along with this.

Reply:
The intuition is that each target result is forced before producing the full result. More precisely, I believe

```haskell
over' p f xs = forceElements res `seq` res
  where
    forceElements = foldlOf' p (\_ !_ -> ()) ()
    res = over p f xs
```

Reply:
That "law" only works with a non-type-changing traversal, but I think it's a good way to express the concept.